src/olympia/api/tests/test_pagination.py

```diff
@@ -54,36 +57,43 @@ def test_metadata_with_default_page_size(self):
 
 class TestESPageNumberPagination(TestCustomPageNumberPagination):
     def test_next_page_never_exeeds_max_result_window(self):
+        total = 50000
         mocked_qs = mock.MagicMock()
-        mocked_qs.__getitem__().execute().hits.total = 30000
+        mocked_qs.__getitem__().execute().hits.total = total
 
         view = generics.ListAPIView.as_view(
             serializer_class=PassThroughSerializer,
             queryset=mocked_qs,
             pagination_class=ESPageNumberPagination,
         )
 
-        request = self.factory.get('/', {'page_size': 5, 'page': 4999})
+        # Request the last page that still has a `next`.
+        page_size = 5
+        page = int(settings.ES_MAX_RESULT_WINDOW / page_size) - 1
+        request = self.factory.get('/', {'page_size': page_size, 'page': page})
         response = view(request)
         assert response.data == {
-            'page_size': 5,
-            'page_count': 5000,
+            'page_size': page_size,
+            'page_count': int(ceil(settings.ES_MAX_RESULT_WINDOW / page_size)),
             'results': mock.ANY,
-            'previous': 'http://testserver/?page=4998&page_size=5',
-            'next': 'http://testserver/?page=5000&page_size=5',
-            'count': 30000,
+            'previous': f'http://testserver/?page={page - 1}&page_size={page_size}',
+            'next': f'http://testserver/?page={page + 1}&page_size={page_size}',
+            'count': total,
         }
 
-        request = self.factory.get('/', {'page_size': 5, 'page': 5000})
+        # Request the page that doesn't have a `next` because it's over the
+        # max result window.
+        page = int(settings.ES_MAX_RESULT_WINDOW / page_size)
+        request = self.factory.get('/', {'page_size': page_size, 'page': page})
         response = view(request)
         assert response.data == {
-            'page_size': 5,
-            'page_count': 5000,
+            'page_size': page_size,
+            'page_count': int(ceil(settings.ES_MAX_RESULT_WINDOW / page_size)),
```

Comment:
this should be just `'page_count': page`?

Reply:
89ee89001f692455fd71a4c50744afe52be2d1fc